packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -165,6 +173,13 @@ export default function TypographyPanel( {
 	panelId,
 	defaultControls = DEFAULT_CONTROLS,
 } ) {
+	// If `panelId` has a value, it means this panel will be rendered inside
+	// the block sidebar. The text alignment UI for individual blocks is rendered
+	// in the block toolbar, so disable support if it's not in the global style.
+	if ( panelId ) {
+		settings.typography.textAlign = false;
+	}
```

Comment:
Rather than relying on fudging the `settings.typography.textAlign` value, could we leverage the fact that this typography panel is extended by the site editor's [global styles typography panel](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/typography-panel.js?plain=1#L18)? 

Perhaps the text alignment control could be rendered only where it is needed that way.

Reply:
> I can't think of any other good approach at the moment, do you have any ideas?

Only the suggestion in my last comment. Did you explore adding the global styles only control as a child to the global styles typography panel?

It makes sense to me that if we have a component extending the typography panel for the site editor's global styles panel that something specific to that is added there instead of hacking the settings in an unexpected manner.

Reply:
> Did you explore adding the global styles only control as a child to the global styles typography panel?

The method I came up with is to pass the context to `StylesTypographyPanel` in the block screen as shown below.

```diff
--- a/packages/edit-site/src/components/global-styles/screen-block.js
+++ b/packages/edit-site/src/components/global-styles/screen-block.js
@@ -260,6 +260,7 @@ function ScreenBlock( { name, variation } ) {
                                        value={ style }
                                        onChange={ setStyle }
                                        settings={ settings }
+                                       context="global-styles"
                                />
                        ) }
                        { hasDimensionsPanel && (
```

And depending on this context, the typography panel controls the UI.

```diff
diff --git a/packages/block-editor/src/components/global-styles/typography-panel.js b/packages/block-editor/src/components/global-styles/typography-panel.js
index 3106723945..bbd10e97b2 100644
--- a/packages/block-editor/src/components/global-styles/typography-panel.js
+++ b/packages/block-editor/src/components/global-styles/typography-panel.js
@@ -173,7 +173,11 @@ export default function TypographyPanel( {
        settings,
        panelId,
        defaultControls = DEFAULT_CONTROLS,
+       context,
 } ) {
+       if ( context === 'global-styles' ) {
+               // Do something...
+       }
        const decodeValue = ( rawValue ) =>
                getValueFromVariable( { settings }, '', rawValue );
```

Is this your intended approach? Sorry if I misunderstood.